crates/accelerate/src/commutation_checker.rs

```diff
@@ -0,0 +1,781 @@
+            .get(&(first_op.name().to_string(), second_op.name().to_string()))
+        {
+            Some(CommutationLibraryEntry::Commutes(b)) => Some(*b),
+            Some(CommutationLibraryEntry::QubitMapping(qm)) => qm
+                .get(&CommutationChecker::get_relative_placement(
+                    first_qargs,
+                    second_qargs,
+                ))
+                .copied(),
+            _ => None,
+        }
+    }
+}
+
+impl ToPyObject for CommutationLibrary {
+    fn to_object(&self, py: Python) -> PyObject {
+        self.library.to_object(py)
+    }
+}
+
+#[pymethods]
+impl CommutationLibrary {
+    #[new]
+    fn new(py: Python, py_any: Option<Py<PyAny>>) -> Self {
+        match py_any {
+            Some(pyob) => {
+                CommutationLibrary { library: pyob.bind(py).extract::<HashMap<(String, String), CommutationLibraryEntry>>().expect("Input parameter standard_gate_commutations to CommutationChecker was not a dict with the right format")}
+
+            },
+            None => CommutationLibrary { library: HashMap::new() }
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum CommutationLibraryEntry {
+    Commutes(bool),
+    QubitMapping(HashMap<SmallVec<[Option<Qubit>; 2]>, bool>),
+}
+
+impl<'py> FromPyObject<'py> for CommutationLibraryEntry {
+    fn extract_bound(b: &Bound<'py, PyAny>) -> Result<Self, PyErr> {
+        if let Ok(b) = b.extract::<bool>() {
+            return Ok(CommutationLibraryEntry::Commutes(b));
+        }
+        let dict = b.downcast::<PyDict>()?;
+        let mut ret = hashbrown::HashMap::with_capacity(dict.len());
+        for (k, v) in dict {
+            let raw_key: SmallVec<[Option<u32>; 2]> = k.extract()?;
+            let v: bool = v.extract()?;
+            let key = raw_key.into_iter().map(|key| key.map(Qubit)).collect();
+            ret.insert(key, v);
+        }
+        Ok(CommutationLibraryEntry::QubitMapping(ret))
+    }
+}
+
+impl ToPyObject for CommutationLibraryEntry {
+    fn to_object(&self, py: Python) -> PyObject {
+        match self {
+            CommutationLibraryEntry::Commutes(b) => PyBool::new_bound(py, *b).to_object(py),
+            CommutationLibraryEntry::QubitMapping(qm) => {
+                let out_dict = PyDict::new_bound(py);
+
+                qm.iter().for_each(|(k, v)| {
+                    out_dict
+                        .set_item(
+                            PyTuple::new_bound(
+                                py,
+                                k.iter()
+                                    .map(|q| q.map(|t| t.0))
+                                    .collect::<Vec<Option<u32>>>(),
+                            )
+                            .to_object(py),
+                            PyBool::new_bound(py, *v).to_object(py),
+                        )
+                        .ok()
+                        .unwrap()
+                });
+                out_dict.to_object(py)
+            }
+        }
+    }
```

Comment:
You don't need to call `.to_object(py)` here after converting to a Python native value if you're sending it back to Python. 

Also, if you possess ownership of the item you're returning, calling `into_py(py)` is better to get the `PyObject` instance as it would allow you to consume the value without unnecessarily copying.

```suggestion
        match self {
            CommutationLibraryEntry::Commutes(b) => b.into_py(py),
            CommutationLibraryEntry::QubitMapping(qm) => {
                let out_dict = PyDict::new_bound(py);

                qm.iter().for_each(|(k, v)| {
                    out_dict
                        .set_item(
                            PyTuple::new_bound(
                                py,
                                k.iter()
                                    .map(|q| q.map(|t| t.0))
                                    .collect::<Vec<Option<u32>>>(),
                            ),
                            PyBool::new_bound(py, *v),
                        )
                        .ok()
                        .unwrap()
                });
                out_dict.unbind().into_any()
            }
        }
```